types/src/transaction/result.rs

```diff
@@ -40,6 +40,7 @@ pub struct Execution<T> {
 
 impl<T> Execution<T> {
     pub fn assert_success(self) -> T {
+        #[allow(clippy::unwrap_used)]
```

Comment:
Intentional unwraps. Should be fine. Though, we can use panic here instead.

Reply:
I would keep using unwrap as it would likely provide more context.